### PR TITLE
JAMES-2704 Avoid listing mailboxes AT ALL in RandomStoring

### PR DESCRIPTION
**Pull request by chibenwa**
Great performance enhancement.

Also, I reworked the test lifecycle, which was not worth the few second gain, especially given JIT the second test is waaay faster.

### CODE REVIEW

File: server/mailet/integration-testing/src/test/java/org/apache/james/smtp/SmtpRandomStoringTest.java [hunk 1]
```diff
@@ -210,14 +202,17 @@ private void checkMailboxesHaveBeenFilled(Collection<IMAPMessageReader> connecti
...
     private LongStream numberOfAUserMessages(IMAPMessageReader imapMessageReader) {
-        return MAILBOXES
-            .stream()
+        return userMailboxesNames()
             .mapToLong(mailbox -> numberOfMessagesInMailbox(imapMessageReader, mailbox));
     }
 
+    private Stream<String> userMailboxesNames() {
+        return IntStream.range(0, PER_USER_MAILBOX_COUNT)
+            .mapToObj(i -> RandomStoring.PROVISIONNED_MAILBOX_PREFIX + i);
```
Comment (mbaechler) [inline server/mailet/integration-testing/src/test/java/org/apache/james/smtp/SmtpRandomStoringTest.java]: you are going to generate mailboxes with too specific names. As we use this mailet for benchmarking it's not a good idea to use a pattern like this for names. However you could generate real random names and always use the same seed to ensure you don't need to store the result of random generation.

File: server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/RandomStoring.java [hunk 2]
```diff
@@ -87,62 +88,51 @@ public String getMailetInfo() {
...
+            .map(user -> new ReroutingInfos(chooseRandomMailboxName(), user))
             .collect(Guavate.toImmutableSet());
     }
 
-    private List<ReroutingInfos> retrieveReroutingInfos() throws UsersRepositoryException {
-        return Streams.stream(usersRepository.list())
-            .map(User::fromUsername)
-            .flatMap(this::buildReRoutingInfos)
-            .collect(Guavate.toImmutableList());
+    private String chooseRandomMailboxName() {
```
Comment (mbaechler) [inline server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/RandomStoring.java]: your "share nothing" design is based on the assumption you know exactly the names of mailboxes expected by the tests. It breaks as soon as you don't use these not-so-random mailbox names.

**Comment by chibenwa** [comment]
The fact that we want mailbox names not to be matching a predefine patterns nor handled by this mailet makes this proposal irrelevant.

**Comment by mbaechler** [comment]
> The fact that we want mailbox names not to be matching a predefine patterns nor handled by this mailet makes this proposal irrelevant.

We could pass the list of mailboxes as a configuration of the mailet, don't you think?

**Comment by chibenwa** [comment]
> We could pass the list of mailboxes as a configuration of the mailet, don't you think?

Not really realistic for a large mailbox count.

**Pull request closed without merging**